lib/mayaUsd/ufe/trf/XformOpUtils.cpp

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright 2024 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "XformOpUtils.h"
+
+#include <maya/MMatrix.h>
+#include <maya/MTransformationMatrix.h>
+#include <maya/MVector.h>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+void extractTRS(const Ufe::Matrix4d& m, Ufe::Vector3d* t, Ufe::Vector3d* r, Ufe::Vector3d* s)
+{
+    // Decompose new matrix to extract TRS.  Neither GfMatrix4d::Factor
+    // nor GfTransform decomposition provide results that match Maya,
+    // so use MTransformationMatrix.
+
+    // We use this function override the default one from UsdUfe(which uses USD API)
+    // to extract the matrix.
+    //
+    //    Example: with input matrix
+    //      [ -0.86, 0, -0.5,  0 ]
+    //      [  0,    1,  0,    0 ]
+    //      [  0.5,  0, -0.86, 0 ]
+    //      [  0,   20,  0,    1 ]
+    //
+    //      USD returns:  [ 180, 30, 180 ]
+    //      Maya returns: [ 0,  150,   0 ]
```

Comment:
This is the reason I kept the Maya way and we set it for MayaUsd.

Reply:
Interesting, IDK if it is due to axis order or if going through some intermediary form like quaternion would give the same results. I say this because if 3dsMax use the USD version, then we will get different result when edited in different DCCs. They would probably like to get the same result we do.

Reply:
Yeah I looked into this with PierreT and we finally discovered that Maya and USD were providing the same visual result, but different value results. Pierre even has a comment about this in the existing code:
https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/ufe/UsdTransform3dUndoableCommands.cpp#L34

So that is why I created the DCC function to the TRS from matrix, which I set for Maya (to keep same result we always had).